R/filter_last_loc.R

```diff
@@ -0,0 +1,23 @@
+#' Find the last location for each individual
+#'
+#' @param x A dataframe containing collar data.
+#' @param id A character vector corresponding the the column to be used as a grouping variable.
+#'
+#' @return A dataframe with the last location for each individual.
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'   filter_last_loc(collar_data)
+#' }
+filter_last_loc <- function(x,
+                            id = "id"
+) {
+
+  assertthat::assert_that(id %in%  names(x), msg = paste(id, "is not a column name in the data frame supplied"))
+
+  x %>%
+  dplyr::group_by(id) %>%
+  dplyr::slice(dplyr::n()) %>%
+  dplyr::ungroup()
+}
```

Comment:
This assumes that the points are ordered by date or something else relevant, so maybe 

```R
x %>%
  dplyr::group_by(id) %>%
  filter(Date == max(Date, na.rm = T))
```

Or if you want to avoid working with dates then 
```R
  x %>%
    dplyr::group_by(id) %>%
   dplyr::arrange(desc(dt_col)) %>%
    dplyr::slice(dplyr::n()) %>%
    dplyr::ungroup()
```

The latter might suggest adding another argument, which I try to resist, but it is not a big deal.

Reply:
I could also see people wanting to do this by other grouping variables. Maybe passing a grouping argument would make it more useful????

```R
filter_last_loc <- function(x,
                            id = "id",
                            grps = NULL,
) {

  assertthat::assert_that(id %in%  names(x), msg = paste(id, "is not a column name in the data frame supplied"))

 grp_vars <- rlang::enquos(grps)

  x %>%
  dplyr::group_by(!!!grp_vars) %>%
  dplyr::slice(dplyr::n()) %>%
  dplyr::ungroup()
}
```

Not sure if I nailed the example, but hope you get the idea just the same.

Reply:
I think that it is good to keep it general and not require the date column to be a date/time class. I will update with the ability to group by and arrange by multiple columns in the next push.